js/main.js

```diff
@@ -678,65 +678,44 @@ function DragAndDropTable(table,mode,queryKey) {
-				window.scrollTo(scrollPos.left, scrollPos.top);
-			}
-		};
+		const isPopoverSupported = () => HTMLElement.prototype.hasOwnProperty("popover");
+		if (!isPopoverSupported) return;
+		els = (typeof els == "object" || typeof els == "function") && typeof els.length == "number" ? els : [els];
+		let popFigureRaw;
+		const imgTemplEl = document.getElementById("tmpl-img-popover");
+		if (imgTemplEl != null) {
+			const popOverTempl = document.importNode(imgTemplEl.content, true);
+			popFigureRaw = popOverTempl.querySelector("figure");
+		} else {
+			popFigureRaw = document.createElementWithAttributes("figure", {"id": "", "className": "full-size-img", "popover": ""});
+			const imgImgTemplate = document.createElementWithAttributes("img", {"src": "", "width": "", "height": "", "alt": ""});
+			popFigureRaw.appendChild(imgImgTemplate);
+		}
 		
-		var oldOnKeyPressFunc = window.document.onkeypress;
-		window.document.onkeypress = function(e) { 
-			if (e.key == "Esc") {
-				imageCanvas.setVisible(false);
-				stopTrigger();
-			}
-			if (typeof oldOnKeyPressFunc == "function")
-				oldOnKeyPressFunc(e);
-		};
-		imageCanvas.onclick = function(e) {
-			imageCanvas.setVisible(false);
-			stopTrigger();
-		}; 
-		imageCanvas.setVisible(false);
-		var fullSizeImage = document.getElementById("fullSizeImage") || document.createElementWithAttributes("img", {"id": "fullSizeImage"}, imageCanvas);
-		for (var i=0; i<els.length; i++) {
-			var links = els[i].getElementsByTagName("a");
-			for (var j=0; j<links.length; j++) {
-				if(links[j].rel.search(/thumbnail/) != -1) {
-					links[j].onclick = function(e) {
-						window.location.hash="image";
-   						var currentHash = window.location.hash;
-						fullSizeImage.src = this.href;
-						imageCanvas.setVisible(true);
-						var imgPoSi = document.getElementPoSi(fullSizeImage);
-						var scrollPos = document.getScrollPosition();
-						var winSize = document.getWindowSize();
-						imageCanvas.style.height=winSize.pageHeight+"px";
-						fullSizeImage.style.marginTop = (scrollPos.top+(winSize.windowHeight-imgPoSi.height)/2) + "px";
-						
-						hashTrigger = window.setInterval(
-							function() {
-								if ( this.location.hash != currentHash ) {
-									imageCanvas.setVisible(false);
-								}
-							},50 
-						);
-						return false;
-					};
+		for (let el of els) {
+			const pID = el.id;
+			const thumbnails = el.querySelectorAll('a[rel=thumbnail]');
+			if (thumbnails.length > 0) {
+				let i = 1;
+				for (thumb of thumbnails) {
+					const postingBody = el.querySelector("div.body");
+					const imgEl = thumb.querySelector("img");
+					const poTarget = [pID, "-img", i].join('');
+					
+					const popFigure = popFigureRaw.cloneNode(true);
+					popFigure.setAttribute("id", poTarget);
+					popFigure.querySelector("img").setAttribute("src", imgEl.getAttribute("src"));
+					postingBody.appendChild(popFigure);
+					imgEl.addEventListener("load", (event) => {
```

Comment:
I kindly request that you refrain from using lambdas, as they render the code illegible.

Reply:
I don't know how to solve this in a meaningful way. The event listener is necessary to ensure the full size image to be loaded. The code in the anonyme function does nothing else than taking the intrinsic sizes and providing them to the image in the popover. I do not expect to reuse this code anywhere else in the foreseeable future. That is why I am in favour of leaving the code as it is.

Reply:
I'm thinking of something like that:

```
imgEl.addEventListener("load", function(event) { 
// do something
});
```

Reply:
Ok, if that's the way to go, I'll give it a go. 🙂